app/upload/views.py

```diff
@@ -15,11 +15,22 @@ def upload_document(service_id):
     if 'document' not in request.files:
         return jsonify(error='No document upload'), 400
 
+    is_csv = False
+    if 'is_csv' in request.form:
```

Comment:
Please note that we are using a form parameter here instead of a JSON post body.  This is not ideal since we have to do some string value checking (below) instead of using a more intuitive boolean value for `is_csv`.

I thought this would be preferable though as it avoids a major change to the API from supporting `multipart/form-data` content types to `application/json`.

It would be good to get some opinions on this.

Reply:
hey @currycoder, thanks for flagging this.

Totally happy if you'd rather not, and we'll still accept the work. That said, if you've got the headspace i'd prefer we make the major change to `application/json`. Reason being, it's an existing pattern we already use (base64 encoded binary data within a json blob) in a few places:

* [our public api for doc download](https://docs.notifications.service.gov.uk/rest-api.html#upload-your-file) (as you already know)
* [our public api for precompiled PDF letters](https://docs.notifications.service.gov.uk/rest-api.html#send-a-precompiled-letter)
* [our internal PDF generation service](https://github.com/alphagov/notifications-template-preview/blob/8b2ae8c78ab912b4c07dcb4873da686af03d628f/app/precompiled.py#L145-L152)

the code could probably look something like

```
if request.is_json():
    file_data = b64decode(request.json['document'])
    is_csv = request.json['is_csv']
else:
    file_data = request.files['document']
    is_csv = False

if not is_instance(is_csv, bool):
    return error... 

[...]
```

then the only change to your existing deployment path for this is changing document_download.py's upload_document to send `json=` instead of `files=` - and we'll need to change that anyway to add the is_csv flag.

that said, if you'd rather not, i'm more than happy to accept this as is - we can do the work to switch over in the future, next time we work on document download.

Reply:
Very neat, thank you.  I'm happy either way on this so wanted to flag it and defer to judgement from you folks here.  Agreed that it makes sense to go with JSON especially since it's an established pattern across repos.

I'll fold in these changes as suggested in a fixup commit and tidy up the commits once we get to approval.

Reply:
This is ready for re-review now :raised_hands: